### PR TITLE
[dap] Don't use --start-paused when running in Profile/Release mode

### PR DESCRIPTION
**Pull request by DanTup**
This change mirrors a fix made in the legacy VS Code extension DAP for https://github.com/Dart-Code/Dart-Code/issues/3838.

The issue is that we were using `--start-paused` in profile mode which would cause a short delay when new isolates were spawned (because they would start paused and wait for us to unpause them). For some measurements of time (including a simple stopwatch) this could make code that uses isolates appear to be slower than it really is.

There was no need to use `--start-paused` for profile mode since debugging functionality is disabled (so we are not concerned about races in setting up breakpoints, etc.) so this simple skips it in those cases.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart [hunk 1]
```diff
@@ -85,30 +85,55 @@ class FlutterDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments
...
-  /// on whether a user chooses to "Run" or "Debug" their app.
+  /// For debugging to be enabled, the user must have chosen "Debug" (and not
+  /// "Run") in the editor (which maps to the DAP `noDebug` field) _and_ must
+  /// not have requested to run in Profile or Release mode. Profile/Release
+  /// modes will always disable debugging.
   ///
-  /// This is always enabled for attach requests, but can be disabled for launch
-  /// requests via DAP's `noDebug` flag. If `noDebug` is not provided, will
-  /// default to debugging.
+  /// This is always `true` enabled for attach requests.
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart:93]: nit. this sentence is awkward. should it be "This is always `true` for attach requests."?

Comment (DanTup) [reply packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart]: Oops! Awkward is an understatement 😄  Fixed!

File: packages/flutter_tools/test/src/dap/flutter_adapter_test.dart [hunk 2]
```diff
@@ -14,8 +14,77 @@ import 'mocks.dart';
 void main() {
   group('flutter adapter', () {
     setUpAll(() {
-    Cache.flutterRoot = '/fake/flutter';
-  });
+      Cache.flutterRoot = '/fake/flutter';
```
Comment (christopherfujino) [inline packages/flutter_tools/test/src/dap/flutter_adapter_test.dart:17]: Is this test actually running on CI? I just noticed that it's under `flutter_tools/test/src`

Comment (DanTup) [reply packages/flutter_tools/test/src/dap/flutter_adapter_test.dart:17]: Hmm, good question! Is `test/src` here not tests for things in `src`? (in [analysis_server](https://github.com/dart-lang/sdk/tree/main/pkg/analysis_server/test/src/computer) it is, so perhaps I assumed incorrectly here).

I can't see which bots are actually running these tests at all - I'll push a change to temporarily break one of these and see if it fails the bot...

Comment (DanTup) [reply packages/flutter_tools/test/src/dap/flutter_adapter_test.dart:17]: I realised afterwards that the tests here are all split into shard folders. You were correct that they were not running. I've moved them to `general.shard` and confirmed they are running now (some are failing on Windows due to paths - I'll push a fix shortly).

**Comment by DanTup** [comment]
@christopherfujino there doesn't seem to be much info about the failure here:

https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20framework_tests_misc/13487/overview

File: packages/flutter_tools/test/src/dap/flutter_adapter_test.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/test/src/dap/flutter_adapter_test.dart]: Yeah, the way our tests are run on CI is very confusing. This map defines all the named shards that we define elsewhere, and then mapping to functions that know where the test files are: https://github.com/flutter/flutter/blob/master/dev/bots/test.dart#L199

`package:flutter_tools` probably needs some master config file declaring where all the test files are, that we could then parse and then verify that each file ending in `_test.dart` is in one of those directories.

**Comment by christopherfujino** [comment]
> https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20framework_tests_misc/13487/overview

I'm guessing this was an infra failure, the swarming task page just says "bot died" https://chromium-swarm.appspot.com/task?id=593f4101f7b6cb10&o=true&w=true

**Review by christopherfujino — APPROVED** [review]
LGTM. the analysis error will be fixed by rebasing upstream.

**Pull request merged**